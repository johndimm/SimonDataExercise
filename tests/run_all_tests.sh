# php
./sendmail_php.sh

# python
./sendmail_python.sh
./render_jinja2.sh

# node
./sendmail_node.sh
./render_handlebars.sh
