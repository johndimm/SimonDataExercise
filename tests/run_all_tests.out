cat sendmail_php.sh

echo "sendmail using php"
export QUERY_STRING="to=jdimm@yahoo.com&from=john.r.dimm@gmail.com&message=hi"; \
php -e -r 'parse_str($_SERVER["QUERY_STRING"], $_GET); include "../jquery/sendmail.php";'
sendmail using php

202
Array
(
    [0] => HTTP/1.1 202 Accepted
    [1] => Server: nginx
    [2] => Date: Mon, 16 Dec 2019 05:15:23 GMT
    [3] => Content-Length: 0
    [4] => Connection: keep-alive
    [5] => X-Message-Id: lCeOxsFNS8SBkgrI7kVOnQ
    [6] => Access-Control-Allow-Origin: https://sendgrid.api-docs.io
    [7] => Access-Control-Allow-Methods: POST
    [8] => Access-Control-Allow-Headers: Authorization, Content-Type, On-behalf-of, x-sg-elas-acl
    [9] => Access-Control-Max-Age: 600
    [10] => X-No-CORS-Reason: https://sendgrid.com/docs/Classroom/Basics/API/cors.html
    [11] => 
    [12] => 
)

cat sendmail_python.sh

echo "sendmail using python"
curl "http://35.231.58.42/cgi-bin/sendEmail.py?message=hi&from=john.r.dimm@gmail.com&to:jdimm@yahoo.com"
sendmail using python
202
b''
Server: nginx
Date: Mon, 16 Dec 2019 05:15:27 GMT
Content-Length: 0
Connection: close
X-Message-Id: 3HDBCy6BSVi6O549Rv8IWw
Access-Control-Allow-Origin: https://sendgrid.api-docs.io
Access-Control-Allow-Methods: POST
Access-Control-Allow-Headers: Authorization, Content-Type, On-behalf-of, x-sg-elas-acl
Access-Control-Max-Age: 600
X-No-CORS-Reason: https://sendgrid.com/docs/Classroom/Basics/API/cors.html


cat render_jinja2.sh

echo "template rendering in python using jinja2"
curl "http://35.231.58.42/cgi-bin/renderTemplate.py?template=hello%%name%%&name=Joshua"
template rendering in python using jinja2
helloJoshua
cat sendmail_node.sh

echo "sendmail in node"
curl "http://localhost:3000/sendMail?message=hi&from=john.r.dimm@gmail.com&to=jdimm@yahoo.com"
sendmail in node
202 Accepted, Mail sent successfully
echo ""
cat render_handlebars.sh

echo "render in node using handlebars"
curl "http://localhost:3000/renderTemplate?template=hello\{\{name\}\}&name=Kitty"
render in node using handlebars
helloKitty